tests/integration/json-queries.js

```diff
@@ -0,0 +1,602 @@
+          numberField: 1.5,
+          nullField: null,
+          booleanField: false,
+          arrayField: [
+            { noMoreLevels: true },
+            1,
+            true,
+            null,
+            "string in jsonObject.arrayField[4]"
+          ],
+          objectField: {
+            object: "string in jsonObject.objectField.object"
+          }
+        },
+        jsonArray: [
+          {
+            stringField: "string in jsonArray[0].stringField",
+            numberField: 5.5,
+            nullField: null,
+            booleanField: true,
+            arrayField: [
+              { noMoreLevels: true }
+            ],
+            objectField: {
+              object: "I'm string in jsonArray[0].objectField.object"
+            }
+          },
+          null,
+          1,
+          "string in jsonArray[3]",
+          false,
+          [
+            { noMoreLevels: true },
+            null,
+            1,
+            "string in jsonArray[5][3]",
+            true
+          ]
+        ]
+      };
+
+      complexJsonObj.jsonObject.jsonArray = _.cloneDeep(complexJsonObj.jsonArray);
+
+      return Promise
+        .all([
+          BoundModel.query().insert(complexJsonObj),
+          BoundModel.query().insert({ id:2, name: "empty object and array", jsonObject: {}, jsonArray: [] }),
+          BoundModel.query().insert({ id:3, name: "null object and array"}),
+          BoundModel.query().insert({ id:4, name: "empty object and [1,2]",   jsonObject: {}, jsonArray: [1,2] }),
+          BoundModel.query().insert({ id:5, name: "empty object and array [ null ]",   jsonObject: {}, jsonArray: [ null ] }),
+          BoundModel.query().insert({ id:6, name: "{a: 1} and empty array", jsonObject: {a: 1}, jsonArray: [] }),
+          BoundModel.query().insert({
+            id: 7,
+            name: "{a: {1:1, 2:2}, b:{2:2, 1:1}} for equality comparisons",
+            jsonObject: {a: {1:1, 2:2}, b:{2:2, 1:1}}, jsonArray: []
+          })
+        ]);
+    });
+
+    it('should have test data', function () {
+      return BoundModel.query().then(function (all) {
+        expect(_.find(all, {name : 'complex line'}).jsonObject.stringField).to.be(complexJsonObj.jsonObject.stringField);
+      });
+    });
+
+    describe('private function parseFieldExpression(expression, extractAsText)', function () {
+      it('should quote ModelJson.jsonArray column reference properly', function () {
+        expect(BoundModel.query().whereJsonIsArray("ModelJson.jsonArray").toString())
+          .to.contain('"ModelJson"."jsonArray"');
+      });
+
+      it('should quote ModelJson.jsonArray:[10] column reference properly', function () {
+        expect(BoundModel.query().whereJsonIsArray("ModelJson.jsonArray:[50]").toString())
+          .to.contain('"ModelJson"."jsonArray"');
+      });
+    });
+
+    describe('.whereJsonObject(fieldExpr, operator, <array|object|string>)', function () {
```

Comment:
All successful tests of this is divided to `.whereJsonEquals`, `.whereJsonSupersetOf` and `.whereJsonSubsetOf`. This was also created when refactoring functionality and didn't want to test failures in every derived method separately. Looks like I'm also calling wrong function in test implementations `whereJsonEquals ` instead of `whereJsonObject` :)